ArduCopter/mode_auto.cpp

```diff
@@ -43,14 +43,31 @@ bool ModeAuto::init(bool ignore_checks)
         // clear guided limits
         copter.mode_guided.limit_clear();
 
-        // start/resume the mission (based on MIS_RESTART parameter)
-        mission.start_or_resume();
+        // don't start the mission until we have an origin
+        Location loc;
+        if (copter.ahrs.get_origin(loc)) {
```

Comment:
I wonder if copter.position_ok() won't be a better check ?

Reply:
I had the thought, but I couldn't prove it was as safe for all use cases, whereas the origin was a much smaller patch which resolved the observed bug and could be shown to be safe.